### DISCUSSION TITLE
dvc-git command pairs

### DISCUSSION THREAD

**Discussion by casperdcl** [treeverse/dvc#5932] [New Features & Ideas]
Probably a duplicate but:

1. make a list of `dvc` & `git` commands which are often run together, e.g.
  - `git checkout <ID> && dvc checkout`
  - `dvc add <file> && git add .gitignore <file>.dvc`
2. discuss ways to make running such command pairs easier
  - (easy) alias certain commands (not really command pairs)
    - `git config --global alias.run '!dvc run' && complete -F git-run _dvc_run`
  - (hard) `dvc` will also run `git` commands?
    - `dvc --git`/`gdvc`/`dvcg`/`gvc` will also run `git` commands?
    - `git config --global alias.dvc '!dvc --git'`
    - `complete -F _dvc git-dvc`

The idea is that e.g. `dvc --git checkout <ID>` or (imho better) `git dvc checkout <ID>` would do both checkouts.

Related: `fds` https://dagshub.com/blog/fds-fast-data-science-with-git-and-dvc/

**Comment by efiop** [comment]
Related https://github.com/iterative/dvc/issues/2363 https://github.com/iterative/dvc/issues/1551

**Comment by jorgeorpinel** [comment]
Also related: #5896
